gaps/gap-4_comp_manifest/gap-4_comp_manifest.md

```diff
@@ -0,0 +1,311 @@
+add the public key of an entity to their trusted key storage. The storage is 
+initialized with a default set of trusted keys, including the one of 
+Golem Factory.
+
+### GCRS `golem.srv.comp.manifest` namespace
+
+Intended for overriding an external Manifest. Allows defining new Manifests
+from the ground up using properties only.
+
+1. `golem.srv.comp.manifest.version : String`
+
+    Specifies a version (Semantic Versioning 2.0 specification) of the 
+    manifest, **defaults** to "0.1.0"
+
+2. `golem.srv.comp.manifest.script` (sub-namespace)
+
+    Defines a set of allowed ExeScript commands and applies constraints to
+    their arguments.
+
+
+3. `golem.srv.comp.manifest.script.commands : List[String]`
+
+    Specifies a curated list of commands in form of:
+
+    - UTF-8 encoded strings
+
+      No command context or matching mode need to be specified.
+
+      E.g. `["run /bin/cat /etc/motd", "run /bin/date -R"]`
+
+    - UTF-8 encoded JSON strings
+
+      Command context (e.g. `env`) or argument matching mode need to be
+      specified for a command.
+
+      E.g. `["{\"run\": { \"args\": \"/bin/date -R\", \"env\": { \"MYVAR\": \"42\" }, \"match\": \"strict\" }}"]`
+
+    - mix of both
+
+    `"deploy"`, `"start"` and `"terminate'` commands are always allowed.
+    These values become the **default** if no `manifest.script.command` 
+    property has been set in the Demand, but the `manifest` namespace is 
+    present.
+
+4. `golem.srv.comp.manifest.script.match : String`
+
+    Selects a default way of comparing command arguments stated in the 
+    manifest and the ones received in the ExeScript, unless stated 
+    otherwise in a command JSON object.
+
+    The `match` property could be one of:
+
+      - `"strict"`: byte-to-byte argument equality (**default**)
+      - `"regex"`: treat arguments as regular expressions
+
+        Syntax: Perl-compatible regular expressions (UTF-8 Unicode mode),
+        w/o the support for look around and backreferences (among others);
+        for more information read the documentation of the Rust
+        [regex](https://docs.rs/regex/latest/regex/) crate.
+
+5. `golem.srv.comp.manifest.net` (sub-namespace)
+
+    Applies constraints to networking. Currently, outgoing requests to the
+    public Internet network are covered.
+
+6. `golem.srv.comp.manifest.net.inet.out.protocols : List[String]`
+
+    List of allowed outbound protocols. Currently **fixed at** `["http", "https"]`.
+
+7. `golem.srv.comp.manifest.net.inet.out.urls : List[String]`
+
+    List of allowed external URLs that outbound requests can be sent to.
+    E.g. `["http://golemfactory.s3.amazonaws.com/file1", "http://golemfactory.s3.amazonaws.com/file2"]`
+
+#### Example
+
+```json
+{
+  "golem.srv.comp.manifest.script.match": "regex",
+  "golem.srv.comp.manifest.script.commands": [
+    "run /bin/cat /etc/motd",
+    "{\"run\": { \"args\": \"/bin/date -R\", \"env\": { \"MYVAR\": \"42\", \"match\": \"strict\" }}}"
```

Comment:
```suggestion
    "{\"run\": { \"args\": \"/bin/date -R\", \"env\": { \"MYVAR\": \"42\" }, \"match\": \"strict\" }}"
```
I guess this was the original intention? At least, this version is in `golem.srv.comp.manifest.script.commands` description, and `match` is described as a `way of comparing command arguments`, not env variables.

Although maybe regexp env variables would also be useful?
Or maybe `match` should be a general attribute that might be somehow added to all possible attributes? E.g.
```
"golem.srv.comp.manifest.net.inet.out" = {
    "urls": ["http://golemfactory.s3.amazonaws.com/file*"],
    "match": "regex",
}
```

Reply:
@mfranciszkiewicz please adjust the GAP content as per the chat on TC call

Reply:
Done